File: GenerateEd25519KeypairDerive/GenerateEd25519KeysDerivation.java
package Generate_Ed25519_key;

import java.security.SecureRandom;
import java.util.Base64;

public class GenerateEd25519KeysDerivation {
    public static void main(String[] args) throws Exception {
        System.out.println("Generate ED25519 private and public key and derive public key from private key");

        TweetNaclFast.Signature.KeyPair keyPair = generateEd25519KeyPair();
        byte[] privateKey = keyPair.getSecretKey();
        byte[] publicKey = keyPair.getPublicKey();
        System.out.println("privateKey (Base64): " + base64Encoding(privateKey));
        System.out.println("publicKey (Base64):  " + base64Encoding(publicKey));

        System.out.println("\nderive the publicKey from the privateKey");
        byte[] publicKeyDerived = deriveEd25519PublicKey(privateKey);
        System.out.println("publicKey (Base64):  " + base64Encoding(publicKeyDerived));
    }

    private static TweetNaclFast.Signature.KeyPair generateEd25519KeyPair() {
        SecureRandom secureRandom = new SecureRandom();
        byte[] key = new byte[64];
        secureRandom.nextBytes(key);
        TweetNaclFast.Signature.KeyPair keyPair = TweetNaclFast.Signature.keyPair_fromSecretKey(key);
        return keyPair;
    }

    private static byte[] deriveEd25519PublicKey(byte[] privateKey) {
        TweetNaclFast.Signature.KeyPair keyPair = TweetNaclFast.Signature.keyPair_fromSecretKey(privateKey);
        return keyPair.getPublicKey();
    }

    private static String base64Encoding(byte[] input) {return Base64.getEncoder().encodeToString(input);}
}

File: GenerateEd25519KeypairDerive/GenerateEd25519KeysDerivation.dart
import 'dart:convert';
import 'dart:typed_data';
import 'package:flutter_sodium/flutter_sodium.dart';

main() {
/* add in pubspec.yaml:
dependencies:
  flutter_sodium: ^0.2.0
 */

  printC('Generate ED25519 private and public key and derive public key from private key');

  KeyPair keyPair = generateEd25519KeyPair();
  Uint8List privateKey = keyPair.sk;
  Uint8List publicKey = keyPair.pk;
  printC('privateKey (Base64): ' + base64Encoding(privateKey));
  printC('publicKey (Base64):  ' + base64Encoding(publicKey));

  printC('\nderive the publicKey from the privateKey');
  Uint8List publicKeyDerived = deriveEd25519PublicKey(privateKey);
  printC('publicKey (Base64):  ' + base64Encoding(publicKeyDerived));
}

KeyPair generateEd25519KeyPair(){
  return CryptoSign.randomKeys();
}

Uint8List deriveEd25519PublicKey(Uint8List privateKey) {
  return CryptoSign.extractPublicKey(privateKey);
}

String base64Encoding(Uint8List input) {
  return base64.encode(input);
}

void printC(String newString) {
  // for compatibility reasons to FlutterEmptyConsole
  print(newString);
}